pulp_ansible/tests/functional/api/collection/v3/test_collection_naming_edgecases.py

```diff
@@ -0,0 +1,168 @@
+        pid = subprocess.run(f"sha256sum {fp}", shell=True, stdout=subprocess.PIPE)
+        checksum = pid.stdout.decode("utf-8").split()[0]
+        return checksum
+
+    # make a content dir
+    content_dir = os.path.join(basedir, "content")
+    os.makedirs(content_dir)
+
+    # make various files
+    for dname in ["docs", "plugins", "roles"]:
+        os.makedirs(os.path.join(content_dir, dname))
+    with open(os.path.join(content_dir, "README.md"), "w") as f:
+        f.write("")
+    with open(os.path.join(content_dir, "plugins", "README.md"), "w") as f:
+        f.write("")
+
+    # make files.json
+    fdata = {"files": [], "format": 1}
+    for root, dirs, files in os.walk(content_dir):
+        for dirname in dirs:
+            dirpath = os.path.join(root, dirname)
+            relative_dir = dirpath.replace(content_dir + "/", "")
+            fdata["files"].append(
+                {
+                    "name": relative_dir,
+                    "ftype": "dir",
+                    "chksum_type": None,
+                    "chksum_sha256": None,
+                    "format": 1,
+                }
+            )
+        for filen in files:
+            filepath = os.path.join(root, filen)
+            relative_file = filepath.replace(content_dir + "/", "")
+            fdata["files"].append(
+                {
+                    "name": relative_file,
+                    "ftype": "file",
+                    "chksum_type": "sha256",
+                    "chksum_sha256": file_checksum(filepath),
+                    "format": 1,
+                }
+            )
+    with open(os.path.join(content_dir, "FILES.json"), "w") as f:
+        f.write(json.dumps(fdata, indent=2))
+
+    # make manifest.json
+    mdata = {
+        "format": 1,
+        "collection_info": {
+            "namespace": namespace,
+            "name": name,
+            "version": version,
+            "authors": ["your name <example@domain.com>"],
+            "readme": "README.md",
+            "tags": [],
+            "description": "your collection description",
+            "license": ["GPL-2.0-or-later"],
+            "license_file": None,
+            "dependencies": {},
+            "repository": "http://example.com/repository",
+            "homepage": "http://example.com",
+            "issues": "http://example.com/issue/tracker",
+        },
+        "file_manifest_file": {
+            "name": "FILES.json",
+            "ftype": "file",
+            "chksum_type": "sha256",
+            "chksum_sha256": file_checksum(os.path.join(content_dir, "FILES.json")),
+            "format": 1,
+        },
+    }
+    with open(os.path.join(content_dir, "MANIFEST.json"), "w") as f:
+        f.write(json.dumps(mdata, indent=2))
+
+    # make the tarball
+    cmd = "tar czvf artifact.tar.gz *"
+    subprocess.run(cmd, shell=True, cwd=content_dir)
+    return os.path.join(content_dir, "artifact.tar.gz")
+
+
+def test_collection_named_collections(
```

Comment:
```suggestion
@pytest.mark.parallel
def test_collection_named_collections(
```
With the randomized name, we should be safe running these tests in parallel with others.